### DISCUSSION TITLE
My predict_model for unseen data not working

### DISCUSSION THREAD

**Discussion by defulamingge** [pycaret/pycaret#2524] [Q&A]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the develop branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@develop).

### Issue Description

Please kindly help I use Python pycaret module to analyze big set of data. I did setup,compare_model,create_model correctly, but when I try to use model I created to predict the unseen_date I splite from beginning, there is only one row come, there is supposee 100k row need predict. I do skip the tune part cause it is take too long but I dont think thats the reason

TSLASAMPLE = TSLA.sample(frac=0.8)
data_unseen  = TSLA.drop(TSLASAMPLE.index)
TSLASAMPLE.reset_index(drop=True, inplace=True)
data_unseen .reset_index(drop=True, inplace=True)
TSLAinput = setup(data = TSLASAMPLE, target= 'prtPrice', use_gpu=True,html=False,silent=True)
dt = create_model('dt')
prediction = predict_model(dt,data=data_unseen)
output: Model MAE MSE RMSE R2 RMSLE MAPE 0 Decision Tree Regressor 0.1842 1.8393 1.3562 0.9996 0.0303 0.0082

### Reproducible Example

```python
pd.options.display.max_columns = 30
path = r'C:\Users\zzmlv\Desktop\758\project' 
all_files = glob.glob(os.path.join(path , "*.csv"))

li = []

for filename in all_files:
    df = pd.read_csv(filename, index_col=None, header=0)
    li.append(df)

frame = pd.concat(li, axis=0, ignore_index=True)
```

### Expected Behavior

0	1.53	Ideal	E	SI1	ID	ID	AGSL	12791	12262.949782
1	1.50	Fair	F	SI1	VG	VG	GIA	10450	10122.442382
2	1.01	Good	E	SI1	G	G	GIA	5161	5032.520456
3	2.51	Very Good	G	VS2	VG	VG	GIA	34361	34840.379469
4	1.01	Good	I	SI1	VG	VG	GIA	4238	4142.695964

### Actual Results

```python-traceback
output:

  Model MAE MSE RMSE R2 RMSLE MAPE
0 Decision Tree Regressor 0.1842 1.8393 1.3562 0.9996 0.0303 0.0082
```

### Installed Versions

<details>
Replace this line with the output of the version code above.
</details>
newest version

**Comment by ngupta23** [comment]
This is not a bug, the individual outputs are in your `prediction` variable. What you see (single row) is the cumulative metric on the unseen data.

Also, I do not know what TSLA contains, but if it is time series data, you should not be doing random sampling as it will not preserve the temporal correlations in the data.
